### PR TITLE
Update Qt

### PR DESCRIPTION
**Pull request by DavidMansolino**
**Description**
A new version of Qt is available for MSYS

### CODE REVIEW

**Comment by DavidMansolino** [comment]
:heavy_check_mark:: travis and appveyor release

**Comment by omichel** [comment]
:eyes:

File: projects/vehicles/plugins/robot_windows/automobile_window/GeneralInformationWidget.hpp [hunk 1]
```diff
@@ -52,7 +52,7 @@ public slots:
   void init();
 
   QCheckBox *mEnableCheckBox;
-  QTime mLastRefreshTime;
+  QElapsedTimer mLastRefreshTime;
```
Comment (omichel) [inline projects/vehicles/plugins/robot_windows/automobile_window/GeneralInformationWidget.hpp]:
```suggestion
  QElapsedTimer mLastRefreshTimer;
```

File: src/webots/app/WbApplication.hpp [hunk 2]
```diff
@@ -20,6 +20,7 @@
 // instanciated/deleted by either WbGuiApplication or WbConsoleApplication
 //
 
+#include <QtCore/QElapsedTimer>
```
Comment (omichel) [inline src/webots/app/WbApplication.hpp]: The definition is not needed here, we should simply declare the class:
```
class QElapsedTimer;
```

and remove the QTime class declaration.

File: src/webots/core/WbGuiRefreshOracle.hpp [hunk 3]
```diff
@@ -46,7 +46,7 @@ class WbGuiRefreshOracle : public QObject {
   static WbGuiRefreshOracle *cInstance;
 
   bool mCanRefreshNow;
-  QTime mLastRefreshTime;
+  QElapsedTimer mLastRefreshTime;
```
Comment (omichel) [inline src/webots/core/WbGuiRefreshOracle.hpp]:
```suggestion
  QElapsedTimer mLastRefreshTimer;
```

File: src/webots/engine/WbSimulationWorld.hpp [hunk 4]
```diff
@@ -72,7 +74,7 @@ protected slots:
   WbOdeContext *mOdeContext;
   WbPhysicsPlugin *mPhysicsPlugin;
   QTimer *mTimer;
-  QTime mLastRealTime;
+  QElapsedTimer mLastRealTime;
```
Comment (omichel) [inline src/webots/engine/WbSimulationWorld.hpp]:
```suggestion
  QElapsedTimer mRealTimeTimer;
```

File: src/webots/engine/WbSimulationWorld.hpp [hunk 5]
```diff
@@ -19,9 +19,11 @@
 // Description: world with physics/kinematic simulation
 //
 
+#include <QtCore/QElapsedTimer>
 #include <QtCore/QList>
 #include <QtCore/QMutex>
-#include <QtCore/QTime>
+#include <QtCore/QTimer>
```
Comment (omichel) [inline src/webots/engine/WbSimulationWorld.hpp]: Are you sure this is needed?
```suggestion
```

File: src/webots/gui/WbConsole.cpp [hunk 6]
```diff
@@ -183,20 +183,17 @@ WbConsole *WbConsole::instance() {
 
 namespace {
   void odeErrorFunc(int errnum, const char *msg, va_list ap) {
-    QString error;
-    error.vsprintf(msg, ap);
+    QString error = QString::vasprintf(msg, ap);
```
Comment (omichel) [inline src/webots/gui/WbConsole.cpp]:
```suggestion
    const QString error = QString::vasprintf(msg, ap);
```

File: src/webots/gui/WbConsole.cpp [hunk 7]
```diff
@@ -183,20 +183,17 @@ WbConsole *WbConsole::instance() {
...
     emit WbLog::instance()->logEmitted(WbLog::ERROR, QString("ODE Error %1: ").arg(errnum) + error, false);
   }
 
   void odeDebugFunc(int errnum, const char *msg, va_list ap) {
-    QString debug;
-    debug.vsprintf(msg, ap);
+    QString debug = QString::vasprintf(msg, ap);
```
Comment (omichel) [inline src/webots/gui/WbConsole.cpp]:
```suggestion
    const QString debug = QString::vasprintf(msg, ap);
```

File: src/webots/gui/WbView3D.cpp [hunk 8]
```diff
@@ -1554,7 +1554,7 @@ void WbView3D::mousePressEvent(QMouseEvent *event) {
     }
   }
   delete mMousePressTime;
-  mMousePressTime = new QTime(QTime::currentTime());
+  mMousePressTime = new QElapsedTimer();
```
Comment (omichel) [inline src/webots/gui/WbView3D.cpp]:
```suggestion
  mMousePressTimer = new QElapsedTimer();
```

File: src/webots/gui/WbView3D.hpp [hunk 9]
```diff
@@ -123,7 +123,7 @@ protected slots:
   WrCameraProjectionMode mProjectionMode;
   WrViewportPolygonMode mRenderingMode;
   int mRefreshCounter;
-  QTime *mMousePressTime;
+  QElapsedTimer *mMousePressTime;
```
Comment (omichel) [inline src/webots/gui/WbView3D.hpp]:
```suggestion
  QElapsedTimer *mMousePressTimer;
```

File: src/webots/engine/WbSimulationWorld.hpp (hunk 5 above)
Comment (DavidMansolino) [reply src/webots/engine/WbSimulationWorld.hpp]: Yes, we are still using a QTimer in this class (but  maybe a forward declaration is sufficient, I will check).

**Pull request merged**